sqlmesh/core/context.py

```diff
@@ -2227,6 +2219,17 @@ def engine_adapters(self) -> t.Dict[str, EngineAdapter]:
                 self._engine_adapters[gateway_name] = adapter
         return self._engine_adapters
 
+    @cached_property
+    def default_catalog_per_gateway(self) -> t.Dict[str, str]:
+        """Returns the catalogs for each engine adapter in a multi virtual layer setup when the catalog isn't shared."""
```

Comment:
> in a multi virtual layer setup

wait, why is it only virtual layer? Shouldn't we apply the model gateway's default catalog to the model specifying that gateway?